### DISCUSSION TITLE
Tanix W2 emmc install

### DISCUSSION THREAD

**Discussion by Orefie** [devmfc/debian-on-amlogic#10] [General]
Hi Devmfc,
Thanks you for your great work.
I tried to install your debian in emmc on a Tanix W2.
With usb key that's work fine, but i have an issue when i start the script for install on emmc.
Can you help me?
Ps: hdmi works fine.

Motherboard:
CS_905W2_TX_2LP4_V1.0

Components
CPU: S905W2
RAM: J256M32LM4
EMMC: H26M52103FMR
WIFI: K255B-SR
ETHERNET: S16037G
LED CONTROLLER: FD650B-S

attach files
[Tanix_parted -l.txt](https://github.com/devmfc/debian-on-amlogic/files/13196165/Tanix_parted.-l.txt)
[Tanix_dmesg.txt](https://github.com/devmfc/debian-on-amlogic/files/13195888/Tanix_dmesg.txt)
[tanix aml-install-to-emmc.txt](https://github.com/devmfc/debian-on-amlogic/files/13195891/tanix.aml-install-to-emmc.txt)

![1698501691247](https://github.com/devmfc/debian-on-amlogic/assets/115935914/2916ab24-3b95-445a-90bc-6edcaa6b7be7)
![1698501691254](https://github.com/devmfc/debian-on-amlogic/assets/115935914/3c95f731-1153-48bb-9c47-416f0be9515c)
![1698501691260](https://github.com/devmfc/debian-on-amlogic/assets/115935914/64f836ed-185c-4960-b574-62daede47aa3)
![1698501691274](https://github.com/devmfc/debian-on-amlogic/assets/115935914/ec078498-e8db-4d52-b15e-eea1ca1a1932)
![1698501691280](https://github.com/devmfc/debian-on-amlogic/assets/115935914/918b2953-742b-41b5-8535-554fc4f04b25)

**Comment by kerta1n** [comment]
https://github.com/devmfc/debian-on-amlogic/discussions/6#discussioncomment-7329760
Try getting root and pulling everything from `/sys/firmware`

**Reply by Orefie** [reply]
Hi @kerta1n 
No problem, I have unpacked the Tanix rom.
But I don't understand the purpose of this operation.
Can you explain?

I see the emmc, but i can't write on it.

![Putty1](https://github.com/devmfc/debian-on-amlogic/assets/115935914/49147f45-3002-4afd-9833-a7adf6c7ece3)

**Reply by devmfc** [reply]
I think that is because the box-config for the TanixW2 has 2 blkdevparts defined. Just comment the line `bootargs9=blkdevparts=mmcblk1:0x2000000@0x018c00000(oem),0x030e400000@0x09d400000(rootfs)` in */boot/box-config/box-tanixw2.config*. After that try again.

**Reply by Orefie** [reply]
HI @devmfc 
After the modification still the same problem.

![Putty2](https://github.com/devmfc/debian-on-amlogic/assets/115935914/f972b9be-ffd5-47b1-a0a0-1731afa1833c)

**Reply by devmfc** [reply]
for some reason it looks like the eMMC is read-only... strange.. What is the output of `/sys/block/mmcblk1/force_ro` And if that is 1, you could try to `echo 0 > /sys/block/mmcblk1/force_ro` and try agian. But if that 'hw' partition is read-only than it will probably be read-only again after a reboot. In that case you can't run a normal debian image on that box. Only workaround that might work in that case is to disable read-only in the bootloader before you boot the kernel.

**Comment by Orefie** [comment]
force_ro is set to 0 but still as read-only. I tried to modify your script. I can now write on mmcblk1p1 but not on  mmcblk1p2.
I also added exfatprogs, else I can't format the mmcblk1p1 as fat. 
I also changed: `dd if=/root/vendor-aml-uboot-backup.img of=${target_disk} conv=noerror bs=512 skip=1 seek=1` and some other things.
`tvbox:~#mkdir -p /media/tmp/dev/mmcblk1p2/{dev,etc,home,mnt,opt,proc,root,run,selinux,srv,sys,tmp,usr,var}
mkdir: cannot create directory ‘/media/tmp/dev’: Read-only file system
mkdir: cannot create directory ‘/media/tmp/dev’: Read-only file system
mkdir: cannot create directory ‘/media/tmp/dev’: Read-only file system
mkdir: cannot create directory ‘/media/tmp/dev’: Read-only file system
mkdir: cannot create directory ‘/media/tmp/dev’: Read-only file system
mkdir: cannot create directory ‘/media/tmp/dev’: Read-only file system
mkdir: cannot create directory ‘/media/tmp/dev’: Read-only file system
mkdir: cannot create directory ‘/media/tmp/dev’: Read-only file system
mkdir: cannot create directory ‘/media/tmp/dev’: Read-only file system
mkdir: cannot create directory ‘/media/tmp/dev’: Read-only file system
mkdir: cannot create directory ‘/media/tmp/dev’: Read-only file system
mkdir: cannot create directory ‘/media/tmp/dev’: Read-only file system
mkdir: cannot create directory ‘/media/tmp/dev’: Read-only file system
mkdir: cannot create directory ‘/media/tmp/dev’: Read-only file system
`
[aml-install-to-emmc.txt](https://github.com/devmfc/debian-on-amlogic/files/13198863/aml-install-to-emmc.txt)

**Reply by devmfc** [reply]
But you can write to /dev/mmcblk1p1 (after formatting with fat and mounting of course)?

**Reply by Orefie** [reply]
yep,
![Capture2](https://github.com/devmfc/debian-on-amlogic/assets/115935914/08e12485-7a2f-48df-8e12-63643bc6ca1d)
![Capture1](https://github.com/devmfc/debian-on-amlogic/assets/115935914/cbf6ac4b-73d7-4474-8ee4-8feff5ed54f7)
Now, I have an issue with the copy on mmcblk1p2 and your script.
But it's moving forward. ;)

**Reply by Orefie** [reply]
I've tried modifying the dtb, changing the emmc frequency,

partition the disk step by step, etc.

But the disk is very unstable.

I can read and write on the disk, but randomly, i have I/O problems.

`
[ 1403.996314] I/O error, dev mmcblk1, sector 246816 op 0x1:(WRITE) flags 0x100000 phys_seg 1 prio class 2
[ 1404.157701] I/O error, dev mmcblk1, sector 247440 op 0x1:(WRITE) flags 0x100000 phys_seg 1 prio class 2
[ 1404.158053] I/O error, dev mmcblk1, sector 247480 op 0x1:(WRITE) flags 0x100000 phys_seg 5 prio class 2
[ 1404.161525] I/O error, dev mmcblk1, sector 247656 op 0x1:(WRITE) flags 0x100000 phys_seg 32 prio class 2
`

in attachments:

The uart log with the latest android rom and the uart log with the debian usb key.

The uart speed is 921600, that's strange. After kernel start, nothing.


If it helps.

[bootlog_tanix_w2_uboot_with_usb_key.txt](https://github.com/devmfc/debian-on-amlogic/files/13259227/bootlog_tanix_w2_uboot_with_usb_key.txt)
[bootlog_tanix_w2_android_rom_2_0230322.txt](https://github.com/devmfc/debian-on-amlogic/files/13259228/bootlog_tanix_w2_android_rom_2_0230322.txt)

**Reply by devmfc** [reply]
That serial speed 921600 is normal for S4/SC2.  
It is also normal not to see anything on serial console after the start of the kernel. If you enable `debug=y` in boot.config you should see more in the serial console.
